File: solutions/62/main.py
from typing import List 

class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for i in range(n)] for j in range(m)]
        # initialize
        for i in range(m):
            dp[i][0] = 1
        for j in range(n):
            dp[0][j] = 1;
        for i in range(1, m):
            for j in range(1, n):
                dp[i][j] = dp[i-1][j] + dp[i][j-1]
        return dp[m-1][n-1]

if __name__ == "__main__":
    s = Solution()
    m, n = 3, 7 
    assert s.uniquePaths(m, n) == 28

File: solutions/62/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Thu Nov 11 08:16:54 2021
> Description:   
 ************************************************************************/
#include <vector> 
#include <gtest/gtest.h>

using namespace std; 

class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n, 0));
        // initialize
        for (int i = 0; i < m; ++i) {
            dp[i][0] = 1;
        }
        for (int j = 0; j < n; ++j) {
            dp[0][j] = 1;
        }
        for (int i = 1; i < m; ++i) {
            for (int j = 1; j < n; ++j) {
                dp[i][j] = dp[i-1][j] + dp[i][j-1];
            }
        }
        return dp[m-1][n-1];
    }
};

TEST(Test62, SimpleTest) {
    const int m = 3; 
    const int n = 7; 
    Solution s; 
    EXPECT_EQ(s.uniquePaths(m, n), 28);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}
